include/openspace/navigation/pathcurve.h

```diff
@@ -0,0 +1,99 @@
+/*****************************************************************************************
+ *                                                                                       *
+ * OpenSpace                                                                             *
+ *                                                                                       *
+ * Copyright (c) 2014-2021                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#ifndef __OPENSPACE_CORE___PATHCURVE___H__
+#define __OPENSPACE_CORE___PATHCURVE___H__
+
+#include <ghoul/glm.h>
+#include <vector>
+
+namespace openspace::interaction {
+
+class Waypoint;
+
+class PathCurve {
+public:
+    virtual ~PathCurve() = 0;
+
+    const double length() const;
+
+    /*
+     * Compute and rteturn the position along the path at the specified relative
+     * distance. The input parameter should be in range [0, 1], where 1 correspond to
+     * the full length of the path
+     */
+    glm::dvec3 positionAt(double relativeDistance);
+
+    /*
+     * Compute curve parameter u that matches the input arc length s
```

Comment:
This comment doesn't seem to match its function. The input is `u` but the comment says it's `s`, which is confusing.